jerry-core/include/jerryscript-snapshot.h

```diff
@@ -28,11 +28,25 @@ extern "C"
  */
 
 /**
+ * Array item for merging snapshots.
+ */
+typedef struct
+{
+  const uint8_t *data_p; /**< snapshot data */
+  size_t size; /**< snapshot size */
+} snapshot_merge_item_t;
+
+/**
  * Snapshot functions.
  */
 size_t jerry_parse_and_save_snapshot (const jerry_char_t *source_p, size_t source_size, bool is_for_global,
                                       bool is_strict, uint32_t *buffer_p, size_t buffer_size);
 jerry_value_t jerry_exec_snapshot (const uint32_t *snapshot_p, size_t snapshot_size, bool copy_bytecode);
+jerry_value_t jerry_exec_snapshot_at (const uint32_t *snapshot_p, size_t snapshot_size,
+                                      size_t func_index, bool copy_bytecode);
+/* Internal API function. Subject to change. */
+size_t jerry_merge_snapshots (const snapshot_merge_item_t *snapshots_p, size_t number_of_snapshots,
+                              uint32_t *buffer_p, size_t buffer_size, const char **error_p);
```

Comment:
I think that the type `snapshot_merge_item_t` is not really needed, the signature could be simply
```c
size_t
jerry_merge_snapshots (uint8_t **in_buffer_pp, /**< array of (pointers to start of) input buffers */
                       size_t *in_buffer_size_p /**< array of input buffer sizes */
                       size_t in_buffer_cnt, /**< number of input buffers */
                       uint8_t *out_buffer_p, /**< (pointer to start of) output buffer */
                       size_t out_buffer_size, /**< size of output buffer */
```
without need for introduction of any new type. This adds only one more parameter to the signature, which is not significant.

We have no history of adding an error string out parameter for our functions, especially not in the closely related `jerry_parse_and_save_snapshot` function so I'm not sure about this param.

And I'd like to have no internal API comments before the function. This function should be as much a first class citizen of the API as the rest of the snapshot functions.